src/OpenProject.Revit/Entry/AppMain.cs

```diff
@@ -27,46 +27,64 @@ public Result OnStartup(UIControlledApplication application)
+        const string tabName = "OpenProject";
+        const string panelName = "BCF management";
+        application.CreateRibbonTab(tabName);
+        RibbonPanel panel = application.CreateRibbonPanel(tabName, panelName);
 
         // Button Data
-        PushButton browserButton = panel.AddItem(new PushButtonData("OpenProject",
-                                                                     "OpenProject",
-                                                                     Path.Combine(_path, "OpenProject.Revit.dll"),
-                                                                     "OpenProject.Revit.Entry.CmdMain")) as PushButton;
-        // Images and Tooltip
-        if (browserButton != null)
+        RibbonItem browserButtonItem = panel.AddItem(
+          new PushButtonData("OpenProject",
+            "OpenProject",
+            Path.Combine(_path, "OpenProject.Revit.dll"),
+            "OpenProject.Revit.Entry.CmdMain"));
+
+        if (browserButtonItem is PushButton browserButton)
         {
           browserButton.Image = LoadPngImgSource("OpenProject.Revit.Assets.OpenProjectLogo16.png");
           browserButton.LargeImage = LoadPngImgSource("OpenProject.Revit.Assets.OpenProjectLogo32.png");
           browserButton.ToolTip = "OpenProject browser";
         }
 
-        PushButton settingsButton = panel.AddItem(new PushButtonData("Settings",
-                                                                      "Settings",
-                                                                      Path.Combine(_path, "OpenProject.Revit.dll"),
-                                                                     "OpenProject.Revit.Entry.CmdMainSettings")) as PushButton;
+        RibbonItem settingsButtonItem = panel.AddItem(
+          new PushButtonData("Settings",
+            "Settings",
+            Path.Combine(_path, "OpenProject.Revit.dll"),
+            "OpenProject.Revit.Entry.CmdMainSettings"));
 
-        if (settingsButton != null)
+        if (settingsButtonItem is PushButton settingsButton)
         {
           settingsButton.Image = LoadPngImgSource("OpenProject.Revit.Assets.Settings32.png");
           settingsButton.LargeImage = LoadPngImgSource("OpenProject.Revit.Assets.Settings32.png");
           settingsButton.ToolTip = "OpenProject Revit Add-in settings";
         }
       }
-      catch (Exception ex1)
+      catch (Exception exception)
       {
-        MessageBox.Show("exception: " + ex1);
+        MessageBox.Show("exception: " + exception);
         return Result.Failed;
       }
 
       return Result.Succeeded;
     }
 
+    private void ConfigureLogger()
+    {
+      var logFilePath = Path.Combine(
+        ConfigurationConstant.OpenProjectApplicationData,
+        "logs",
+        "OpenProject.Revit.Log..txt");
+
+      Log.Logger = new LoggerConfiguration()
+        .MinimumLevel.Debug()
+        .WriteTo.File(logFilePath, rollingInterval: RollingInterval.Day)
```

Comment:
Maybe we should also set a max size? I'm not sure if that's present implicitly, but I remember on another project where we needed to configure Serilog for files we added a configuration option to set the max log file size.

Reply:
max size is set per default to 1GB, and per default there is no rollover after hitting max size. I considered those values as i.o.